R/get_schema.R

```diff
@@ -59,8 +59,8 @@ get_schema.Id <- function(obj,  ...) {
 #' @export
 #' @rdname get_schema
 get_schema.PqConnection <- function(obj, temporary = FALSE,  ...) {
-  if (isTRUE(temporary))  {
-    return(DBI::dbGetQuery(obj, "SELECT nspname FROM pg_namespace WHERE oid = pg_my_temp_schema();")$nspname)
+  if (temporary)  {
+    return("pg_temp")
```

Comment:
Good point, I guess it does not work because "pg_temp" is an alias and not the **actual** schema of the table; I am getting the same result with `dplyr::tbl` as well.

Although I would expect the footprint of copying `mtcars` to be insignificant, I have seen Redshift handle copying quite slowly. If all we are doing is creating a temporary table so that Postgres assigns a pg_temp_nnn schema, I think this would be a slight improvement:

```suggestion
  if (temporary)  {
    temp_schema <- DBI::dbGetQuery(obj, "SELECT nspname FROM pg_namespace WHERE oid = pg_my_temp_schema();")$nspname

    # If no temporary tables have been created, the temp schema has not been determined yet
    if (identical(temp_schema, character(0))) {
      temp_table_name <- unique_table_name()
      DBI::dbExecute(conn, glue::glue("CREATE TEMPORARY TABLE {temp_table_name}(a TEXT)"))
      temp_table <- dplyr::tbl(conn, temp_table_name)
      defer_db_cleanup(temp_table)
      rm(temp_table_name)

      temp_schema <- DBI::dbGetQuery(obj, "SELECT nspname FROM pg_namespace WHERE oid = pg_my_temp_schema();")$nspname
    }

    return(temp_schema)
```

Reply:
Sounds good. Let me test that implementation instead :)